auth0/src/main/java/com/auth0/android/Auth0.java

```diff
@@ -281,11 +282,17 @@ private HttpUrl resolveConfiguration(@Nullable String configurationDomain, @NonN
         return url;
     }
 
-    private HttpUrl ensureValidUrl(String url) {
+    @VisibleForTesting
+    HttpUrl ensureValidUrl(String url) {
         if (url == null) {
             return null;
         }
-        String safeUrl = url.startsWith("http") ? url : "https://" + url;
+
+        if (url.toLowerCase().startsWith("http://")) {
```

Comment:
we should save this `url.toLowerCase()` in a temporary variable, as 2 lines below we do another `startsWith` that might not match. 
e.g. "Https://myapp.com/" could be constructed into "Https://myapp.com/https://myapp.com"

Reply:
Yes, good catch. New commit pushed to fix and test.